services/explorer/src/pages/TransactionData/TopicDataDisplay.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box, Dropdown, Typography } from '@taraxa_project/taraxa-ui';
+import { useState } from 'react';
+import useStyles from '../../components/DataRow/DataRow.styles';
+
+export interface EventDataProps {
+  title: string;
+  hexValue: string;
+}
+
+export const TopicDataDisplay = ({ title, hexValue }: EventDataProps) => {
+  const classes = useStyles();
+
+  const toggleValues = [
+    { value: 'hex', label: 'Hex' },
+    { value: 'dec', label: 'Dec' },
+  ];
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [value, setValue] = useState<string | null>(null);
+  const open = Boolean(anchorEl);
+
+  const [decoded, setDecoded] = useState<string>(hexValue);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = (val: string) => {
+    setValue(val);
+    setAnchorEl(null);
+    switch (val) {
+      case toggleValues[0].value:
+        setDecoded(hexValue);
+        break;
+      case toggleValues[1].value:
+        const addressHex = '0x' + hexValue.slice(-40);

```

Comment:
I just checked and this is the wrong way of doing it. This converts the address to decimal. However, on a log's topics we have only addresses, so all we need to do is to cut the padded zeroes to make the address readable.

Reply:
Example: https://etherscan.io/tx/0x9051b74a7a2be1c071453f6b0ff076ab0173b07bb647927933c292c85a8f7076#eventlog